src/calendar/day/basic/index.js

```diff
@@ -92,7 +94,7 @@ class Day extends Component {
         accessibilityRole={isDisabled ? undefined : 'button'}
         accessibilityLabel={this.props.accessibilityLabel}
       >
-        <Text allowFontScaling={false} style={textStyle}>{String(this.props.children)}</Text>
+        <Text allowFontScaling={false} style={textStyle}>{String(this.props.formatNumber ? this.props.formatNumber(this.props.children) : this.props.children)}</Text>
```

Comment:
Why duplicate this line through all the Day components instead of doing it once where you send the text in Calendar to the <DayComp> element?